rumqttc/src/client.rs

```diff
@@ -398,8 +397,6 @@ impl<'a> Iterator for Iter<'a> {
 
 impl<'a> Drop for Iter<'a> {
     fn drop(&mut self) {
-        // TODO: Don't create new runtime in drop
-        let runtime = runtime::Builder::new_current_thread().build().unwrap();
-        self.connection.runtime = Some(mem::replace(&mut self.runtime, runtime));
+        self.connection.runtime = None;
```

Comment:
```suggestion
        self.connection.runtime = self.runtime.take();
```

Reply:
yeah that makes more sense, it slipped by mind that mem::replace returns back the old value.